### PR TITLE
Migrate GWC REST API to Spring MVC to keep in sync with GeoServer core

### LINKED ISSUE #532: Update the Rest API to Spring MVC
The rest api for GeoServer has been updated to Spring MVC 4 removing it's dependency on Restlet.  It would be beneficial for GeoWebCache to follow suite and update to Spring MVC as well.

### PR DESCRIPTION
**Pull request by vickdw**
Resolves #532

### CODE REVIEW

**Comment by vickdw** [comment]
Resolves Update the Rest API to Spring MVC #532

File: geowebcache/diskquotarest/pom.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
```
Comment (smithkm) [inline geowebcache/diskquotarest/pom.xml]: Wrong Copyright header

File: geowebcache/pom.xml [hunk 2]
```diff
@@ -229,6 +229,12 @@
       <scope>test</scope>
     </dependency>
     <dependency>
+      <groupId>org.easymock</groupId>
```
Comment (smithkm) [inline geowebcache/pom.xml]: The classextension capability should be standard in EasyMock 3.2.  This artifact should just add a package named `classextension` to make old code work.

File: geowebcache/restng/pom.xml [hunk 3]
```diff
@@ -0,0 +1,148 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (smithkm) [inline geowebcache/restng/pom.xml]: Shouldn't this module have been renamed to `rest`

File: geowebcache/restng/src/main/java/org/geowebcache/restng/controller/GWCController.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Arne Kepp / The Open Planning Project 2009
+ * @author David Vick, Boundless, Copyright 2017
+ *
+ * Original file
+ * https://github.com/GeoWebCache/geowebcache/blob/master/geowebcache/rest/src/main/java/org/geowebcache/rest/GWCRestlet.java
```
Comment (smithkm) [inline geowebcache/restng/src/main/java/org/geowebcache/restng/controller/GWCController.java]: Lets us know the name of the original file, but the link is going to break when we get rid of the old rest module in master.  That it it includes an indication that there's an original file to look for and what it's named is enough for me though.

File: geowebcache/web/src/test/java/org/geowebcache/jetty/RestIT.java [hunk 5]
```diff
@@ -1,44 +1,32 @@
...
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+
 /**
  * Integration test for the REST API in a full GWC instance
  * 
  * @author Kevin Smith, Boundless
  *
  */
+@Ignore
```
Comment (smithkm) [inline geowebcache/web/src/test/java/org/geowebcache/jetty/RestIT.java]: Why ignore these tests? They are black box integration tests of the REST API so they should really be passing after the changes.

File: geowebcache/web/src/main/webapp/WEB-INF/geowebcache-security.xml [hunk 6]
```diff
@@ -16,6 +16,7 @@
     <security:http>
         <security:intercept-url method="GET" pattern="/rest/web/**" access="permitAll"/>
         <security:intercept-url pattern="/rest/**" access="hasRole('ROLE_ADMINISTRATOR')"/>
+        <security:intercept-url pattern="/restng/**" access="hasRole('ROLE_ADMINISTRATOR')"/>
```
Comment (smithkm) [inline geowebcache/web/src/main/webapp/WEB-INF/geowebcache-security.xml]: Again isn't this supposed to be replacing the old REST not sitting beside it?

Changes to context mean changes to GeoServer will be required so we need a PR downstream for that.

**Comment by jodygarnett** [comment]
@smithkm has requested changes, are they now resolved?

File: geowebcache/pom.xml (hunk 2 above)
Comment (vickdw) [reply geowebcache/pom.xml]: I believe this has been addressed in later commits.

**Comment by jodygarnett** [comment]
This build fails on build.geoserver.org, but passes for me locally.

```
Failed tests: 
  RestIT.testCreateUpdateDelete:124 
Expected: hasProperty("statusCode", <200>)
     but: property 'statusCode' was <503>
  RestIT.testDiskQuotaJson:412 expected:<200> but was:<503>
  RestIT.testDiskQuotaXML:382 expected:<200> but was:<503>
  RestIT.testInvalidMethods:241->testSecured:328 
Expected: hasProperty("statusCode", <405>)
     but: property 'statusCode' was <503>
  RestIT.testSecure:258->testSecured:339 
Expected: Request <GET http://localhost:8080/geowebcache/rest/layers.xml HTTP/1.1> with without authentication produces status code <401>
     but: property 'statusCode' was <503>
Tests in error: 
  RestIT.testAddLayer:349->doGetXML:470->doGetXML:479 » SAXParse The element typ...
  RestIT.testGetLayer:264->doGetXML:470->doGetXML:479 » SAXParse The element typ...
  RestIT.testGetLayers:89->doGetXML:470->doGetXML:479 » SAXParse The element typ...
  RestIT.testLayerNoAuth:287->doGetXML:470->doGetXML:479 » SAXParse The element ...
```

Each of the tests seems to have the same problem, expecting a security failure and get a service not available failure.

```
testSecure(org.geowebcache.jetty.RestIT)  Time elapsed: 0.013 sec  <<< FAILURE!
java.lang.AssertionError: 

Expected: Request <GET http://localhost:8080/geowebcache/rest/layers.xml HTTP/1.1> with without authentication produces status code <401>
     but: property 'statusCode' was <503>
	at org.geowebcache.jetty.RestIT.testSecured(RestIT.java:339)
	at org.geowebcache.jetty.RestIT.testSecure(RestIT.java:258)
```

See https://build.geoserver.org/view/geowebcache/job/geowebcache-master/1583/console

**Comment by jodygarnett** [comment]
rerunning build now passes.

**Pull request merged**